src/config.rs

```diff
@@ -34,208 +130,89 @@ pub struct Config {
-                    .default_value("100"),
-            )
-            .arg(
-                Arg::with_name("bulk_index_threads")
-                    .long("bulk-index-threads")
-                    .help("Number of threads used for bulk indexing (default: use the # of CPUs)")
-                    .default_value("0")
-            )
-            .arg(
-                Arg::with_name("tx_cache_size")
-                    .long("tx-cache-size")
-                    .help("Number of transactions to keep in for query LRU cache")
-                    .default_value("10000")  // should be enough for a small wallet.
-            )
-            .arg(
-                Arg::with_name("blocktxids_cache_size")
-                    .long("blocktxids-cache-size")
-                    .help("Number of blocks to cache transactions IDs in LRU cache")
-                    .default_value("100")) // Needs ~0.305MB per per block at 10k txs each
-            .arg(
-                Arg::with_name("txid_limit")
-                    .long("txid-limit")
-                    .help("Number of transactions to lookup before returning an error, to prevent \"too popular\" addresses from causing the RPC server to get stuck (0 - disable the limit)")
-                    .default_value("100")  // should take a few seconds on a HDD
-            )
-            .arg(
-                Arg::with_name("server_banner")
-                    .long("server-banner")
-                    .help("The banner to be shown in the Electrum console")
-                    .default_value(&default_banner)
-            )
-            .get_matches();
-
-        let network_name = m.value_of("network").unwrap_or("mainnet");
-        let network_type = match network_name {
-            "mainnet" => Network::Bitcoin,
-            "testnet" => Network::Testnet,
-            "regtest" => Network::Regtest,
-            _ => panic!("unsupported Bitcoin network: {:?}", network_name),
+        use internal::ResultExt;
+
+        let system_config: &OsStr = "/etc/electrs/config.toml".as_ref();
+        let home_config = home_dir().map(|mut dir| { dir.push(".electrs/config.toml"); dir });
+        let cwd_config: &OsStr = "electrs.toml".as_ref();
+        let configs = std::iter::once(cwd_config)
+            .chain(home_config.as_ref().map(AsRef::as_ref))
+            .chain(std::iter::once(system_config));
+
+        let (mut config, _) = internal::Config::including_optional_config_files(configs).unwrap_or_exit();
+
+        let network_name = match config.network {
+            Network::Bitcoin => "bitcoin",
+            Network::Testnet => "testnet",
+            Network::Regtest => "regtest",
         };
-        let db_dir = Path::new(m.value_of("db_dir").unwrap_or("./db"));
-        let db_path = db_dir.join(network_name);
 
-        let default_daemon_port = match network_type {
+        config.db_dir.push(network_name);
```

Comment:
Damn, I just noticed this will change the name of the dir from `mainnet` to `bitcoin` this is pretty big backwards-incompatibility - going to fix it.